treeprint.go

```diff
@@ -40,6 +46,10 @@ type Tree interface {
 
 	SetValue(value Value)
 	SetMetaValue(meta MetaValue)
+
+	// VisitAll iterates over the tree, branches and nodes.
+	// If need to iterate over the whole tree, use the root node.
+	VisitAll(fn NodeVisitor)
```

Comment:
Might be nice to mention here that this is a depth-first walk.

Reply:
I was never the good regarding algorithm names but since the code is iterating over all items in the current branch first to them head to the next branch, wouldn't this be breadth first? Btw, yeah it could be worth adding that info to the docs, you're right 👍.